nomad/server.go

```diff
@@ -353,6 +353,112 @@ func NewServer(config *Config, consulCatalog consul.CatalogAPI, logger *log.Logg
 	return s, nil
 }
 
+// Start the RPC listeners
+func (s *Server) startRPCListener() {
+	s.rpcListenerLock.Lock()
+	defer s.rpcListenerLock.Unlock()
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
generally I would prefer contexts be passed in so the caller controls the cancel func, but I think for this use-case it's fine